plutus-conformance/add-test-output/Spec.hs

```diff
@@ -61,58 +61,76 @@ runnerReader inp =
-    MkArgs extension directory run <- customExecParser (prefs showHelpOnEmpty) args
+    MkArgs extension directory run fileOpt <- customExecParser (prefs showHelpOnEmpty) args
     allInputFiles <- findByExtension [extension] directory
-    -- only choose the ones without an output file, so as to not edit the ones already with outputs
-    inputFiles <- filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
+    inputFiles <-
+      case fileOpt of
+        Missing -> filterM (fmap (fmap not) (\testIn -> doesFileExist (testIn <> ".expected"))) allInputFiles
+        All     -> pure allInputFiles
     case run of
-      Eval -> do
-        for_ inputFiles
-          (\inputFile -> do
-            inputStr <- readFile inputFile
-            let parsed = runQuoteT $ UPLC.parse UPLC.program inputFile $ T.pack inputStr
-                outFilePath = inputFile <> ".expected"
-            case parsed of
-              Left (ParseErrorB peb) -> do
-                -- warn the user that the file failed to parse
-                putStrLn $ inputFile <> " failed to parse. Error written to " <> outFilePath
-                writeFile outFilePath (errorBundlePretty peb)
-              Right pro -> do
-                res <- try (evaluate $ evalUplcProg (() <$ pro)):: IO (Either SomeException (EvaluationResult UplcProg))
-                case res of
-                  Right (EvaluationSuccess prog) -> do
-                    putStrLn $ inputFile <> " evaluated; result written to " <> outFilePath
-                    writeFile outFilePath (render $ pretty prog)
-                  Right EvaluationFailure      -> do
-                    -- warn the user that the file failed to evaluate
-                    putStrLn $ inputFile <> " failed to evaluate. Failure written to " <> outFilePath
-                    writeFile outFilePath shownEvaluationFailure
-                  Left _ -> do
-                    -- warn the user that exception is thrown
-                    putStrLn $ "Exception thrown during evaluation of " <> inputFile <>".Written to " <> outFilePath
-                    writeFile outFilePath shownEvaluationFailure
-          )
+      Eval ->
+        for_ inputFiles printAndWrite
+          where printAndWrite inputFile = do
+                  inputTxt <- T.readFile inputFile
+                  let parsed = parseTxt inputTxt
+                      outFilePath = inputFile <> ".expected"
+                  case parsed of
+                    Left (ParseErrorB _) -> do -- specifying parsed to ParserError for the compiler
+                      -- warn the user that the file failed to parse
+                      putStrLn $ inputFile <> " failed to parse. Error written to " <> outFilePath
+                      T.writeFile outFilePath shownParseError
+                    Right pro -> do
+                      res <- try (evaluate $ evalUplcProg (() <$ pro)):: IO (Either SomeException (EvaluationResult UplcProg))
```

Comment:
Add a call to `force` before `evaluate` in order for exceptions not to sneak into some thunk. The AST of UPLC is strict, but we can still end up having an `error` in a `Data` value or whatever.